helpers.js

```diff
@@ -0,0 +1,150 @@
+const _ = require("lodash");
+const { access } = require("fs/promises");
+const fs = require("fs");
+
+function prepareRunPlaybookPayload({
+  sshUsername,
+  sshPass,
+  sshKeyPath,
+  playbookPath,
+  inventoryFiles,
+  inventoryIPs: inventoryIps,
+  limit,
+  modules,
+  vars,
+}) {
+  const sshCredentials = {};
+  if (sshUsername?.trim()) {
+    sshCredentials.username = sshUsername.trim();
+  }
+  if (sshPass?.trim()) {
+    sshCredentials.password = sshPass.trim();
+  }
+  if (sshKeyPath?.trim()) {
+    sshCredentials.keyPath = sshKeyPath.trim();
+  }
+
+  return {
+    playbookPath: playbookPath.trim(),
+    sshCredentials,
+    inventoryFiles,
+    inventoryIps,
+    limit,
+    modules,
+    vars: parseVarsParam(vars),
+  };
+}
+
+function parseVarsParam(varsParam) {
+  if (_.isNil(varsParam)) {
+    return {};
+  }
+
+  if (_.isPlainObject(varsParam)) {
+    return varsParam;
+  }
+
+  if (_.isArray(varsParam)) {
```

Comment:
Please reverse the logic to perform the parsing (happy path) at the end of the function block and return early by throwing the error (unhappy path) when the param doesn't meet our expectations.